Library/Homebrew/cask/utils.rb

```diff
@@ -47,7 +47,7 @@ def self.gain_permissions_remove(path, command: SystemCommand)
       gain_permissions(path, permission_flags, command) do |p|
         if p.parent.writable?
           if directory
-            p.rmtree
+            FileUtils.rm_r p
```

Comment:
`Pathname#rmtree` now never returning errors is a horrifically subtle change that questions the safety of every `#rmtree` call we have, given the new behaviour now means `rmtree` can not actually remove the directory.

Reply:
e.g. this error catching now doesn't work:

https://github.com/Homebrew/brew/blob/97030b799262c331ad017f5a20cd2c0b9a13d8d1/Library/Homebrew/reinstall.rb#L73-L78

We could override the behaviour in `extend/pathname.rb`. Seems hacky but might be the safest solution.

Reply:
Yeh, good catch and idea here @Bo98.